### PR TITLE
Allow hostname to be configured

### LINKED ISSUE #224: How to set a custom container hostname?
I want to run the equivalent of the following for [RabbitMQ](https://hub.docker.com/_/rabbitmq):
```
docker run -d --hostname my-rabbit --name some-rabbit rabbitmq:3.8
```
However, I do not see anything in `testcontainers.ContainerRequest` that could map to that option.
All I have at the moment is:
```golang
func TestQueueDeclaration(t *testing.T) {
	ctx := context.Background()
	containerRequest := testcontainers.ContainerRequest{
		Image: "rabbitmq:3.8.5",
		Name:  fmt.Sprintf("test-rabbit-%s", t.Name()),
		ExposedPorts: []string{"5672/tcp"},
		// Hostname???
		WaitingFor: wait.ForLog("Server startup complete"),
	}
	rabbitContainer, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
		ContainerRequest: containerRequest,
		Started:          true,
	})
	assert.NilError(t, err)
	defer func(context context.Context) {
		assert.NilError(t, rabbitContainer.Terminate(context))
	}(ctx)
	host, err := rabbitContainer.Host(ctx)
	assert.NilError(t, err)
	port, err := rabbitContainer.MappedPort(ctx, "5672")
	assert.NilError(t, err)
        rabbitUrl := fmt.Sprintf("amqp://%s:%s", host, port.Port())

	queue, err := // [...]

	assert.NilError(t, err)
	assert.Equal(t, queue.Name, fmt.Sprintf("%s/%s", namespace, queueName))
}
```
Is setting a custom container hostname supported?

### PR DESCRIPTION
**Pull request by fbiville**
Fixes #224

### CODE REVIEW

File: docker_test.go [hunk 1]
```diff
@@ -1265,5 +1268,53 @@ func TestContainerNonExistentImage(t *testing.T) {
...
+			t.Fatal(err)
+		}
+	}()
+	if actualHostname := readHostname(t, container.GetContainerID()); actualHostname != hostname {
+		t.Fatalf("expected hostname %s, got %s", hostname, actualHostname)
+	}
+}
 
+// TODO: replace with proper API call
+func readHostname(t *testing.T, containerId string) string {
```
Comment (fbiville) [inline docker_test.go:1302]: looking for guidance on what proper call should be used here instead 🙏

**Comment by fbiville** [comment]
Any updates on this?

File: container.go [hunk 2]
```diff
@@ -89,6 +89,7 @@ type ContainerRequest struct {
 	AutoRemove      bool                // if set to true, the container will be removed from the host when stopped
 	NetworkMode     container.NetworkMode
 	AlwaysPullImage bool // Always pull image
+	Hostname        string
```
[the remark this answers is not in the document]
Comment (fbiville) [reply container.go]: Done :)

File: docker_test.go [hunk 3]
```diff
@@ -1265,5 +1268,53 @@ func TestContainerNonExistentImage(t *testing.T) {
...
 	})
+}
+
+func TestContainerWithCustomHostname(t *testing.T) {
+	ctx := context.Background()
+	name := fmt.Sprintf("some-rabbit-%s-%d", t.Name(), rand.Int())
+	hostname := fmt.Sprintf("my-rabbit-%s-%d", t.Name(), rand.Int())
+	req := ContainerRequest{
+		Name:     name,
+		Image:    "rabbitmq",
```
[the remark this answers is not in the document]
Comment (fbiville) [reply docker_test.go]: Done :)

**Comment by mdelapenya** [comment]
It seems Travis is complaining about go-fmt. Could you locally check it running same script as in the CI?

```shell
$ ./scripts/checks.sh
```

**Comment by fbiville** [comment]
> It seems Travis is complaining about go-fmt. Could you locally check it running same script as in the CI?
> 
> ```shell
> $ ./scripts/checks.sh
> ```

Thanks, now fixed.

**Pull request merged**